### DISCUSSION TITLE
2021.11.29 희의록 + 멘토링

### DISCUSSION THREAD

**Discussion by hharimjung** [O-per/cakd3_Project3#30] [Meeting Minutes]
## 1129 멘토링 피드백
1. 진척속도가 느림
2. 내일 다시 멘토링 
3. 일지에 나와있는거 그대로 다시

## 29일 모델 진척사항 공유
* ET5(이슬)
  - source code의 py 파일들 파악 중 각각의 클래스, arguments, 함수에 대한 최소한의 이해가 있어야 오류가 왜 어디서 일어나는지 알 수 있을 것 같은데 seq2seq py 제공된 파일 외에도 huggingface의 source code까지 파악해야해서 시간이 오래 걸리는 중
  https://github.com/O-per/cakd3_Project3/discussions/22
  - 허깅페이스 팀에 문의
  - 에트리에 문의 해둔 상태

* ET5(경림)
  - ET5로 모델 training하기 위해 준비중
  - training 및 모델 가중치 저장까지 가능할 것으로 보이나 fine-tuning은 아직임

* ET5(수문)
  - ET5 오류, 버전을 낮춰 보는 등 시도 했으나 수정 실패
  - 현재는 BART 모델 https://github.com/insublee/GAS_summarization 시도중

* 유빈(모델은 아니지만 진척사항 공유)
  - 피그마 사용해서 어느정도 모델 웹 구현 틀 제작
  - html공부 조금,,,함,,,ㅎ
  - 피그마를 플라스크에 연결하여 웹 구현을 할 생각인데 일반적으로 생각하는 챗봇(?)의 모형과 달라서 그 모형을 넣기 위해 정보 수집중 
  - 저희가 지금 하는것이 text rank라는것인가요…? 그것과 비슷하여 text rank 웹구현 소스 찾아보는중

* 하림
  - 피그마로 전반적인 프로그램 툴 작업 및 유빈씨와 PPT 수정작업
  : 사용할 모델이 확정되면 모델과 관련된 기술들 추가예정
  - 깃허브 README 업데이트
  - 생성될 모델을 플라스크 사용하여 연동시키는 방법 서치 중
  - 텍스트요약 모델 프로세스 자료 서치 중
  - 2차 작업파일 기반으로 코드 참고 및 수정방안 탐색

## 회의록
* ET5 VS KET5
* ET5
  - T5 대로 ET5를 구동시키면 작동은 되지만, 파인튜닝한 py 파일을 불러와서 작동시키면 돌아가지않음. 경림님이 py 파일을 돌렸을 때 작동은 됐지만 성능이 너무 좋지않아서 파일 사용이 제대로 되었는지 의심이 되는 상태. (loss 값이 적으면 좋지만, validation 은 8.0~9.0, 좋지않음 / 학습량을 늘리면 loss값이 떨어져야되는데 validation의 갯수를 늘렸을때는 loss값이 왜 증가하는지)
  - 1조의 코드 출처를 알게 됨 (https://shivanandroy.com/fine-tune-t5-transformer-with-pytorch/)  -> T5는 잘 돌아감 / ET5는 메모리 오류가 생기는데 해결방법을 잘 모르겠음
* KET5
  - 학습시킬 데이터 신청 중
* 데이터셋은 아직 연락없음. 사용하지 못한다고 가정하고 PPT 내용 수정할 것(AI HUB 데이터만 사용)
* 개발 계획서 및 요약 보고서 작성 완료
